From d930ba960a1abaf8b76e507707a7fe081a376741 Mon Sep 17 00:00:00 2001
From: Miroslav Rezanina <mrezanin@redhat.com>
Date: Mon, 17 Dec 2018 11:04:15 +0100
Subject: [PATCH 2/2] Do not build bluetooth support

RH-Author: Miroslav Rezanina <mrezanin@redhat.com>
Message-id: <1545044655-7801-1-git-send-email-mrezanin@redhat.com>
Patchwork-id: 83537
O-Subject: [RHEL-7.7 qemu-kvm PATCH] Do not build bluetooth support
Bugzilla: 1654627
RH-Acked-by: Stefan Hajnoczi <stefanha@redhat.com>
RH-Acked-by: Danilo de Paula <ddepaula@redhat.com>
RH-Acked-by: Markus Armbruster <armbru@redhat.com>

From: Miroslav Rezanina <mrezanin@redhat.com>

We do not support bt devices however some bt related code is built in.
As this code is deprecated upstream disabling build of this code downstream.

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 Makefile.objs       | 2 +-
 hw/bt/Makefile.objs | 4 ++--
 qemu-options.hx     | 2 ++
 vl.c                | 6 ++++++
 4 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/Makefile.objs b/Makefile.objs
index 74f722e44e..1712ee6133 100644
--- a/Makefile.objs
+++ b/Makefile.objs
@@ -64,7 +64,7 @@ common-obj-y += audio/
 common-obj-y += hw/
 
 common-obj-y += ui/
-common-obj-y += bt-host.o bt-vhci.o
+#common-obj-y += bt-host.o bt-vhci.o
 
 common-obj-y += dma-helpers.o
 common-obj-y += vl.o
diff --git a/hw/bt/Makefile.objs b/hw/bt/Makefile.objs
index 867a7d2e8a..e678e9ee3c 100644
--- a/hw/bt/Makefile.objs
+++ b/hw/bt/Makefile.objs
@@ -1,3 +1,3 @@
-common-obj-y += core.o l2cap.o sdp.o hci.o hid.o
-common-obj-y += hci-csr.o
+#common-obj-y += core.o l2cap.o sdp.o hci.o hid.o
+#common-obj-y += hci-csr.o
 
diff --git a/qemu-options.hx b/qemu-options.hx
index 24ffab6a50..bcf1b8f755 100644
--- a/qemu-options.hx
+++ b/qemu-options.hx
@@ -2157,6 +2157,7 @@ STEXI
 @end table
 ETEXI
 
+#if 0
 DEFHEADING(Bluetooth(R) options:)
 STEXI
 @table @option
@@ -2231,6 +2232,7 @@ STEXI
 @end table
 ETEXI
 DEFHEADING()
+#endif
 
 #ifdef CONFIG_TPM
 DEFHEADING(TPM device options:)
diff --git a/vl.c b/vl.c
index 7c34b7c64f..5c5cdbe88b 100644
--- a/vl.c
+++ b/vl.c
@@ -840,6 +840,7 @@ static void configure_rtc(QemuOpts *opts)
     }
 }
 
+#if 0 // Disabled for Red Hat Enterprise Linux
 /***********************************************************/
 /* Bluetooth support */
 static int nb_hcis;
@@ -961,6 +962,7 @@ static int bt_parse(const char *opt)
     fprintf(stderr, "qemu: bad bluetooth parameter '%s'\n", opt);
     return 1;
 }
+#endif
 
 static int parse_sandbox(QemuOpts *opts, void *opaque)
 {
@@ -3132,9 +3134,11 @@ int main(int argc, char **argv, char **envp)
                     exit(1);
                 break;
 #endif
+#if 0 // Disabled for Red Hat Enterprise Linux
             case QEMU_OPTION_bt:
                 add_device_config(DEV_BT, optarg);
                 break;
+#endif
             case QEMU_OPTION_audio_help:
                 AUD_help ();
                 exit (0);
@@ -4113,9 +4117,11 @@ int main(int argc, char **argv, char **envp)
     }
 #endif
 
+#if 0 // Disabled for Red Hat Enterprise Linux
     /* init the bluetooth world */
     if (foreach_device_config(DEV_BT, bt_parse))
         exit(1);
+#endif
 
     if (!xen_enabled()) {
         /* On 32-bit hosts, QEMU is limited by virtual address space */
-- 
2.19.1

